### PR TITLE
Add failing test around moduleName mutation

### PR DESCRIPTION
**Pull request by asakusuma**
Failing test that addresses https://github.com/emberjs/ember.js/issues/14192

cc @krisselden @rwjblue @trentmwillis

### CODE REVIEW

File: packages/ember-glimmer/tests/integration/components/local-lookup-test.js [hunk 1]
```diff
@@ -48,6 +48,20 @@ moduleFor('Components test: local lookup', class extends RenderingTest {
     this.assertText('Nested template says: Hi!', 'Re-render works');
   }
 
+  ['@htmlbars moduleName remains unchanged']() {
```
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: There ain't no htmlbars up in here

Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: You will need to target the PR against beta branch to get it to run tests.

Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: Hmm. I'm happy to land this test here though too.  Can you remove the `@htmlbars` prefix, and open a separate PR against beta?

Comment (asakusuma) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: Done

File: packages/ember-glimmer/tests/integration/components/local-lookup-test.js [hunk 2]
```diff
@@ -48,6 +48,20 @@ moduleFor('Components test: local lookup', class extends RenderingTest {
...
   }
 
+  ['@test moduleName remains unchanged']() {
+    this.registerComponent('x-outer/x-inner', { template: 'Nested template says: {{yield}}' });
+    this.registerComponent('x-outer', { template: '{{#x-inner}}Hi!{{/x-inner}}' });
+    this.registerComponent('x-root', { template: '{{x-outer}}' });
+
+    this.render('{{x-root}}');
+
+    let moduleName = this.owner.lookup('template:-top-level').meta.moduleName;
```
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: Looks like `this.owner.lookup('template:-top-level')` is returning `undefined` (causing the Travis failures).

**Comment by asakusuma** [comment]
Ah, so this test is testing an implementation detail that is engine specific. Meta is stored differently between glimmer and htmlbars. I've updated the test, but I'm thinking it should be glimmer only.

**Comment by rwjblue** [comment]
Yeah, we completely removed the htmlbars stuff. So targeting glimmer only is perfect.

File: packages/ember-glimmer/tests/integration/components/local-lookup-test.js [hunk 3]
```diff
@@ -48,6 +48,20 @@ moduleFor('Components test: local lookup', class extends RenderingTest {
...
   }
 
+  ['@test moduleName remains unchanged']() {
+    this.registerComponent('x-outer/x-inner', { template: 'Nested template says: {{yield}}' });
+    this.registerComponent('x-outer', { template: '{{#x-inner}}Hi!{{/x-inner}}' });
+    this.registerComponent('x-root', { template: '{{x-outer}}' });
+
+    this.render('{{x-root}}');
+
+    let moduleName = this.owner.lookup('template:-top-level').spec.meta.moduleName;
```
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/components/local-lookup-test.js]: Maybe we can make moduleName a getter on the template wrapper class...

**Comment by rwjblue** [comment]
@asakusuma - Can you update this test to work with the recent refactor (making meta available both on the factory and the instance)?  We can test that `deepEquals(owner._lookupFactory('template:-top-level').meta, owner.lookup('template:-top-level').meta)` too.

**Comment by rwjblue** [comment]
@asakusuma - Ping?

**Comment by asakusuma** [comment]
@rwjblue sorry I totally forgot about this, yes I will do that.

**Comment by rwjblue** [comment]
@asakusuma - I'm going to go ahead and close this for now, please submit a new PR if you have a chance to pick this up again...

**Pull request closed without merging**